run_test.sh

```diff
@@ -32,36 +38,48 @@ find generated -type f -not \( -name "*.expected" -or -name "__init__.py" \) -de
 )
 
 (
-    # Run mypy (always under python3)
+    # Run mypy
+    eval "$(pyenv init -)"
+    pyenv shell $PY_VER_MYPY
+    PY_VERSION=`python -c 'import sys; print(sys.version.split()[0])'`
+    VENV=venv_$PY_VERSION
 
     # Create virtualenv
-    if [[ -z $SKIP_CLEAN ]] || [[ ! -e $MYPY_VENV ]]; then
+    if [[ -z $SKIP_CLEAN ]] || [[ ! -e $VENV ]]; then
         python3 --version
         python3 -m pip --version
-        python3 -m virtualenv $MYPY_VENV
+        python3 -m virtualenv $VENV
     fi
-    source $MYPY_VENV/bin/activate
-    if [[ -z $SKIP_CLEAN ]] || [[ ! -e $MYPY_VENV ]]; then
+    source $VENV/bin/activate
+    if [[ -z $SKIP_CLEAN ]] || [[ ! -e $VENV ]]; then
         python3 -m pip install setuptools
         python3 -m pip install git+https://github.com/python/mypy.git@985a20d87eb3a516ff4457041a77026b4c6bd784
     fi
 
     # Run mypy
-    for PY in 2.7 3.5; do
```

Comment:
notably here - a single run of this script only runs a mypy target version (defaulting to 3.5) - however it will update both output.expected if either fails (for convenience).

This should make it easier to selectively compile grpc proto files only on python3.